purchase_open_qty/init_hook.py

```diff
@@ -75,11 +75,27 @@ def store_field_qty_to_receive_and_invoice(cr):
         """
         UPDATE purchase_order_line
         SET qty_to_receive = pol.qty
-        FROM (SELECT purchase_line_id, sum(product_uom_qty) as qty
-              FROM stock_move
-              WHERE purchase_line_id IS NOT NULL AND
-                  state not in ('cancel', 'done')
-              GROUP BY purchase_line_id) as pol
+        FROM (
+            SELECT
+                pol.id AS purchase_line_id,
+                CASE
+                    WHEN pt.type = 'service' THEN pol.product_qty - pol.qty_received
+                    ELSE (
+                        SELECT SUM(sm.product_uom_qty)
+                        FROM stock_move sm
+                        WHERE sm.purchase_line_id = pol.id
+                        AND sm.state NOT IN ('cancel', 'done')
+                    )
+                END AS qty
+            FROM purchase_order_line pol
+            JOIN product_template pt ON pt.id = (
+                SELECT pt2.id
+                FROM product_product pp2
+                JOIN product_template pt2 ON pt2.id = pp2.product_tmpl_id
+                WHERE pp2.id = pol.product_id
+                LIMIT 1
+            )
```

Comment:
```suggestion
            FROM purchase_order_line pol
            JOIN product_product pp ON pp.id = pol.product_id
           JOIN product_template pt ON pt.id = pp.product_tmpl_id
```
Can you check my suggestion is working well?

Reply:
What you pointed out worked fine. I've updated the code. Thank you!